lib/everypolitician/popolo/collection.rb

```diff
@@ -13,6 +17,27 @@ def -(other)
         other_ids = Set.new(other.documents.map(&:id))
         documents.reject { |d| other_ids.include?(d.id) }
       end
+
+      private
+
+      # TODO: This feels pretty nasty, is there a better way of working out the
+      # class name?
+      def klass
+        case self.class.to_s.split("::").last
+        when "People"
+          Person
+        when "Organizations"
+          Organization
+        when "Memberships"
+          Membership
+        when "Events"
+          Event
+        when "Areas"
+          Area
+        else
+          raise "Unknown class: #{self.class.to_s}"
+        end
```

Comment:
Perhaps it would be better if each collection had to explicitly define it's member class? Something like this:

```ruby
class Collection
  def self.member_class(klass = nil)
    @member_class = klass unless klass.nil?
    @member_class
  end
end

class Areas < Collection
  member_class Area
end
```

Then you could call `self.class.member_class` to get the class.

A couple of observations/thoughs:

- I'm not sure if `member_class` is the right name, any suggestions?
- This would make the collection classes a bit more useful, rather than just being empty as they are now
- This seems like a nice application of [replace conditional with polymorphism](https://sourcemaking.com/refactoring/replace-conditional-with-polymorphism).

What do you think?